src/permission/rule-application.js

```diff
@@ -0,0 +1,420 @@
+var OPEN = 'open';
+var UNDEFINED = 'undefined';
+var LOADING = 'loading';
+var ERROR = 'error';
+var STRING = 'string';
+var EOL = require( 'os' ).EOL;
+
+var C = require( '../constants/constants' );
+var RecordRequest = require( '../record/record-request' );
+var messageParser = require( '../message/message-parser' );
+var JsonPath = require( '../record/json-path' );
+var utils = require( '../utils/utils' );
+
+/**
+ * This class handles the evaluation of a single rule. It creates
+ * the required variables, injects them into the rule function and
+ * runs the function recoursively until either all cross-references,
+ * references to old or new data is loaded, it errors or the maxIterationCount
+ * limit is exceeded
+ *
+ * @constructor
+ *
+ * @param {Object} params requires the following keys
```

Comment:
@param { { username: string, authData: objec  } }

or 
( From usejsdoc )
/**
 * Assign the project to an employee.
 * @param {Object} employee - The employee who is responsible for the project.
 * @param {string} employee.name - The name of the employee.
 * @param {string} employee.department - The employee's department.
 */

etc..